docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,525 @@
+    uint8_t platform_instance_id[16];///< Value of Platform Instance ID, 16 bytes
+    pck_cert_flag_enum_t dynamic_platform; ///< Indicate whether a platform can be extended with additional packages
+                                           ///< via Package Add calls to SGX Registration Backend
+    pck_cert_flag_enum_t cached_keys;      ///< Indicate whether platform root keys are cached by SGX Registration Backend
+    pck_cert_flag_enum_t smt_enabled;      ///< Indicate whether a plat form has SMT (simultaneous multithreading) enabled
+
+} sgx_ql_qv_supplemental_t;
+```
+
+## Proposed Changes
+
+In order to smoothly switch quote verification from OE to SGX DCAP QVL/QvE, we
+propose to implement the change in 2 steps as below:
+- Step 1: Implement QVL-based quote verification and make it co-exist with the
+  existing implementation
+- Step 2: Remove existing implementation and makes the DCAP QVL a hard dependency
+
+### Step 1: Implement QVL-based quote verification and make it co-exist with the existing implementation
+If SGX DCAP QVL/QvE are not installed, quote verification will continue to use existing
+logic. Otherwise, quote verification will switch to QVL/QvE automatically.
+
+#### Add functions in host-side plugin for SGX ECDSA-p256 quote verification
+To align with the current implementation of SGX quote-ex library integration,
+the OE SDK host-side verifier plugin library will dynamically detect the presence of
+QVL library and load it at runtime. In case the QVL library can't be found, a specfic
+error will be returned.
+
+- Add functions in file `host/sgx/sgxquote.c` and `host/sgx/quote.c`
+  - Use environment variable, e.g. “USE_SGX_QVL”. Only when this variable is set,
+    try to detect & load the QVL library, return specific error if environment variable
+    not set or QVL doesn't exist
+  - Call QVL API to verify the quote
+
+#### Add OCALL to support enclave-side SGX ECDSA-p256 quote verification
+As quote verification will be done by DCAP QVL/QvE, for enclave-side plugin,
+it requires OCALL to call host-side APIs.
+
+In this proposal, we will add one OCALL `oe_verify_quote_ocall()` in
+`edl/sgx/attestation.edl`.
+The OCALL is used for passing quote buffer and related parameters to host side.
+All other relevant logic will be implemented on the host side, in files
+`host/sgx/ocalls.c` and `host/sgx/quote.c`.
+If DCAP QVL cannot be found in host-side plugin, a specific error will be returned.
+
+Note that currently only ECDSA-p256 quotes are supported, but the OCALL will keep
+`format_id` and `opt_params` for forward compatibility.
+
+```C
+oe_result_t oe_verify_quote_ocall(
+    [in] const oe_uuid_t* format_id,
+    [in, size=opt_params_size] const void* opt_params,
+    size_t opt_params_size,
+    [in, size=quote_size] const void* p_quote,
+    size_t quote_size,
+    oe_datetime_t expiration_check_date,
+    [out] uint32_t *p_collateral_expiration_status,
```

Comment:
This seems odd to have to do things in 2 passes, one to get untrusted values and then oe_verify_qve_report_and_identity() to verify those values are correct.  I didn't follow why this two-phase process was needed.

Reply:
> This seems odd to have to do things in 2 passes, one to get untrusted values and then oe_verify_qve_report_and_identity() to verify those values are correct. I didn't follow why this two-phase process was needed.

The `untrusted values` you mentioned are QvE output. When verifier uses QvE to verify the SGX quote, he/she needs to confirm 2 things.
- QvE is trusted and up to date
- The outputs are really come from QvE, and they have not been changed

So the 2nd phase is needed for above two items.